src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -6,29 +6,31 @@ namespace Datadog.Trace.ClrProfiler.Managed.Loader
     /// <summary>
     /// A class that attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
     /// </summary>
-    public class Startup
+    public partial class Startup
     {
         /// <summary>
         /// Initializes static members of the <see cref="Startup"/> class.
         /// This method also attempts to load the Datadog.Trace.ClrProfiler.Managed. NET assembly.
         /// </summary>
         static Startup()
         {
-            LoadManagedAssembly();
+            ManagedProfilerDirectory = ResolveManagedProfilerDirectory();
+            AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve_ManagedProfilerDependencies;
+            TryLoadManagedAssembly();
         }
 
-        internal static bool ManagedAssemblyFound { get; set; }
+        internal static string ManagedProfilerDirectory { get; }
 
-        private static void LoadManagedAssembly()
+        private static bool TryLoadManagedAssembly()
         {
             try
             {
                 Assembly.Load(new AssemblyName("Datadog.Trace.ClrProfiler.Managed, Version=1.7.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb"));
```

Comment:
Should we use the assembly version from `DatadogTracer.ClrProfiler.Managed.Loader` here instead of hard-coding it? (assuming they will always match)

```suggestion
                var assemblyName = Assembly.GetExecutingAssembly().GetName();
                assemblyName.Name = "Datadog.Trace.ClrProfiler.Managed";
                
                Assembly.Load(assemblyName);
```

Reply:
That's a good idea, I hadn't thought of that before. The version should always be incremented with the native (and managed) profiler so I'm happy to make this change.

Reply:
Fixed in 7a2a2ce4d640c750eca578a28ef7c32cde1eadb4 .